devel/happy/Portfile

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           haskell_stack 1.0
+
+name                happy
+version             1.19.12
+categories          devel haskell
+maintainers         openmaintainer
+license             GPL-3
+platforms           macosx
+homepage            https://github.com/simonmar/happy
+
+description         Happy is a parser generator for Haskell 98 (and later).
+long_description    ${description}. \
+            Happy is a parser generator system for Haskell, similar to \
+            the tool `yacc' for C. Like `yacc', it takes a file containing \
+            an annotated BNF specification of a grammar and produces a \
+            Haskell module containing a parser for the grammar.  Happy \
+            is flexible: you can have several Happy parsers in the same \
+            program, and several entry points to a single grammar. Happy \
+            can work in conjunction with a lexical analyser supplied by \
+            the user (either hand-written or generated by another program), \
+            or it can parse a stream of characters directly (but this \
+            isn't practical in most cases).
+
+master_sites        https://hackage.haskell.org/package/${name}-${version}
+
+conflicts           hs-happy
+
+checksums           rmd160  34fff4797b87496a0493960b784df96385d4d6c0 \
+                    sha256  fb9a23e41401711a3b288f93cf0a66db9f97da1ce32ec4fffea4b78a0daeb40f \
+                    size    183254
+
+# relative paths to ${prefix}
+set happy_datadir   share/${name}
+
+post-extract {
+    xinstall -o macports -m 0755 -d \
+        "[option haskell_stack.stack_root]" \
+        ${destroot}${prefix}/${happy_datadir}
+
+    # Fix for cabal data-files hardcoded path in binary
+    # See:
+    # https://github.com/commercialhaskell/stack/issues/848
+    # https://github.com/commercialhaskell/stack/issues/4857
+    # https://github.com/haskell/cabal/issues/462
+    # https://github.com/haskell/cabal/issues/3586
+    xinstall -m 0644 -W ${worksrcpath} \
+        ${filespath}/Paths_${name}.hs ./src
+
+    reinplace "s|@PREFIX@|${prefix}|g" \
+        ${worksrcpath}/src/Paths_${name}.hs
+}
+
+post-destroot {
+    # install cabal data-files
+    fs-traverse f ${worksrcpath}/.stack-work {
+        if { [file isdirectory ${f}]
+            && [string match "${name}-${version}" [file tail ${f}]] } {
```

Comment:
Isn't this the same as `[file tail ${f}] eq "${name}-${version}"`?

Reply:
If you say! I’ll change it.